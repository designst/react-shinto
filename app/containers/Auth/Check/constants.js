import createConstant from 'utils/createConstant';

export const AUTH_CHECK_REQUEST = createConstant(__filename, 'AUTH_CHECK_REQUEST');
export const AUTH_CHECK_SUCCESS = createConstant(__filename, 'AUTH_CHECK_SUCCESS');
export const AUTH_CHECK_FAILURE = createConstant(__filename, 'AUTH_CHECK_FAILURE');
