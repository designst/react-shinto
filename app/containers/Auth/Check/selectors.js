import { createSelector } from 'reselect';

const selectAuthDomain = state => state.auth;

export const makeSelectToken = () => createSelector(selectAuthDomain, state => state.token);
