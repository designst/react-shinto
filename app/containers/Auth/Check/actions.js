import { ERROR_ACTION, WAIT_FOR_ACTION } from 'redux-wait-for-action';

import { AUTH_CHECK_REQUEST, AUTH_CHECK_SUCCESS, AUTH_CHECK_FAILURE } from './constants';

export const checkAuthRequest = () => ({
  type: AUTH_CHECK_REQUEST,
});

export const checkAuthSuccess = data => ({
  type: AUTH_CHECK_SUCCESS,
  data,
});

export const checkAuthFailure = error => ({
  type: AUTH_CHECK_FAILURE,
  error,
});

export const checkAuthRequestWait = () => ({
  ...checkAuthRequest(),
  [WAIT_FOR_ACTION]: AUTH_CHECK_SUCCESS,
  [ERROR_ACTION]: AUTH_CHECK_FAILURE,
});
