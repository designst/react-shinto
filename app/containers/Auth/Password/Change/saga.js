import { call, put, takeLatest, getContext } from 'redux-saga/effects';

import createLogger from 'utils/createLogger';

import { AUTH_PASSWORD_CHANGE_REQUEST } from 'containers/Auth/Password/Change/constants';

import {
  passwordChangeSuccess,
  passwordChangeFailure,
} from 'containers/Auth/Password/Change/actions';

const logger = createLogger(__filename);

const authPasswordChangeUrl = process.env.SHINTO_AUTH_PASSWORD_CHANGE_API_ENDPOINT;

export function* authPasswordChange(action) {
  const { oldPassword, newPassword, newPasswordConfirm } = action.payload;

  const apiService = yield getContext('apiService');

  logger('Password Change');

  try {
    const response = yield call(apiService.post, authPasswordChangeUrl, {
      oldPassword,
      newPassword,
      newPasswordConfirm,
    });
    yield put(passwordChangeSuccess(response.data));
  } catch (err) {
    yield put(passwordChangeFailure(err));
  }
}

export default function* authPasswordChangeData() {
  yield takeLatest(AUTH_PASSWORD_CHANGE_REQUEST, authPasswordChange);
}
