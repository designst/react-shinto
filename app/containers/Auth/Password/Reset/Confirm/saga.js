import { call, put, takeLatest, getContext } from 'redux-saga/effects';

import createLogger from 'utils/createLogger';

import { AUTH_PASSWORD_RESET_CONFIRM_REQUEST } from 'containers/Auth/Password/Reset/Confirm/constants';

import {
  passwordResetConfirmSuccess,
  passwordResetConfirmFailure,
} from 'containers/Auth/Password/Reset/Confirm/actions';

const logger = createLogger(__filename);

const authPasswordResetConfirmUrl = process.env.SHINTO_AUTH_PASSWORD_RESET_CONFIRM_API_ENDPOINT;

export function* authPasswordResetConfirm(action) {
  const { token } = action;

  const apiService = yield getContext('apiService');

  logger('Password Reset Confirm: %s', token);

  try {
    const response = yield call(apiService.post, authPasswordResetConfirmUrl, {
      token,
    });
    yield put(passwordResetConfirmSuccess(response.data));
  } catch (err) {
    yield put(passwordResetConfirmFailure(err));
  }
}

export default function* authPasswordResetConfirmData() {
  yield takeLatest(AUTH_PASSWORD_RESET_CONFIRM_REQUEST, authPasswordResetConfirm);
}
