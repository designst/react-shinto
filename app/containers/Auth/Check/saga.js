import { call, put, select, takeLatest, getContext } from 'redux-saga/effects';

import createLogger from 'utils/createLogger';

import { makeSelectToken } from 'containers/Auth/Check/selectors';
import { CHECK_AUTH_REQUEST } from 'containers/Auth/Check/constants';
import { checkAuthSuccess, checkAuthFailure } from 'containers/Auth/Check/actions';

const logger = createLogger(__filename);

const checkAuthUrl = process.env.SHINTO_AUTH_CHECK_API_ENDPOINT;

export function* checkAuth() {
  const token = yield select(makeSelectToken());

  const apiService = yield getContext('apiService');

  logger('Check Auth: %s', token);

  try {
    const data = yield call(apiService.get, checkAuthUrl, { token });
    yield put(checkAuthSuccess(data));
  } catch (err) {
    yield put(checkAuthFailure(err));
  }
}

export default function* checkAuthData() {
  yield takeLatest(CHECK_AUTH_REQUEST, checkAuth);
}
