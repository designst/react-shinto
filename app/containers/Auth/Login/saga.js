import { call, put, takeLatest, getContext } from 'redux-saga/effects';

import createLogger from 'utils/createLogger';

import { AUTH_LOGIN_REQUEST } from 'containers/Auth/Login/constants';
import { loginSuccess, loginFailure } from 'containers/Auth/Login/actions';

const logger = createLogger(__filename);

const authLoginUrl = process.env.SHINTO_AUTH_LOGIN_API_ENDPOINT;

export function* authLogin(username, password) {
  const apiService = yield getContext('apiService');

  logger('Login: %s', username);

  try {
    const response = yield call(apiService.post, authLoginUrl, { username, password });
    yield put(loginSuccess(response.data));
  } catch (err) {
    yield put(loginFailure(err));
  }
}

export default function* authLoginData() {
  yield takeLatest(AUTH_LOGIN_REQUEST, authLogin);
}
