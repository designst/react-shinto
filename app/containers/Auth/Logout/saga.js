import { call, put, takeLatest, getContext } from 'redux-saga/effects';

import createLogger from 'utils/createLogger';

import { AUTH_LOGOUT_REQUEST } from 'containers/Auth/Logout/constants';
import { logoutSuccess, logoutFailure } from 'containers/Auth/Logout/actions';

const logger = createLogger(__filename);

const authLogoutUrl = process.env.SHINTO_AUTH_LOGOUT_API_ENDPOINT;

export function* authLogout() {
  const apiService = yield getContext('apiService');

  logger('Logout');

  try {
    const response = yield call(apiService.get, authLogoutUrl);
    yield put(logoutSuccess(response.data));
  } catch (err) {
    yield put(logoutFailure(err));
  }
}

export default function* authLogoutData() {
  yield takeLatest(AUTH_LOGOUT_REQUEST, authLogout);
}
