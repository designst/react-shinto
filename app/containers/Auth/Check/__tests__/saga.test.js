import { expectSaga } from 'redux-saga-test-plan';

import authReducer from 'containers/Auth/reducer';
import { authCheck } from 'containers/Auth/Check/saga';

describe('authCheckData', () => {
  let authCheckGenerator;

  beforeEach(() => {
    authCheckGenerator = authCheck();

    const selectDescriptor = authCheckGenerator.next().value;
    expect(selectDescriptor).toMatchSnapshot();
  });

  it('should dispatch the authCheckSuccess action if it requests the data successfully', () =>
    expectSaga(authCheck)
      .provide({
        select: () => 'token',
        getContext: () => ({
          post: url => (url === '/' ? {} : {}),
        }),
      })
      .withReducer(authReducer)
      .hasFinalState({
        token: null,
        isAuthenticated: true,
        isAuthenticating: false,
      })
      .run());

  it('should dispatch the authCheckFailure action if it requests the data without token', () =>
    expectSaga(authCheck)
      .provide({
        select: () => null,
        getContext: () => ({
          post: () => ({}),
        }),
      })
      .withReducer(authReducer)
      .run());

  it('should dispatch the authCheckFailure action if it requests the data unsuccessfully', () =>
    expectSaga(authCheck)
      .provide({
        select: () => 'token',
        getContext: () => ({
          post: () => {
            throw new Error('Error');
          },
        }),
      })
      .withReducer(authReducer)
      .run());
});
