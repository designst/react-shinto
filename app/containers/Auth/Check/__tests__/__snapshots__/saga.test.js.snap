// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`authCheckData should dispatch the authCheckFailure action if it requests the data unsuccessfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [],
    "selector": [Function],
  },
}
`;

exports[`authCheckData should dispatch the authCheckFailure action if it requests the data without token 1`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [],
    "selector": [Function],
  },
}
`;

exports[`authCheckData should dispatch the authCheckSuccess action if it requests the data successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [],
    "selector": [Function],
  },
}
`;
